File: LeetCode/1855. Maximum Distance Between a Pair of Values/solution.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        n, m = len(nums1), len(nums2)
        i = j = 0
        res = 0
        while i < n:
            while j < m and nums2[j] >= nums1[i]:
                j += 1
            res = max(res, max(0,j-i-1))
            i += 1
        return res

File: LeetCode/1855. Maximum Distance Between a Pair of Values/solution.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int n1 = nums1.size(), n2 = nums2.size(), res = 0;
        int j = 0;
        for (int i = 0; i < n1; i++){
            while (j < n2 && nums1[i] <= nums2[j]){
                j++;
            }
            res = max(res, max(0,j-1-i));
        }
        return res;
    }
};